include/zephyr/drivers/stepper_motor_controller.h

```diff
@@ -0,0 +1,279 @@
+	stepper_motor_controller_enable_velocity_mode_t
+		stepper_motor_controller_enable_velocity_mode;
+#endif /* CONFIG_STEPPER_MOTOR_CONTROLLER_HAS_CONSTANT_VELOCITY_MODE */
+#ifdef CONFIG_STEPPER_MOTOR_CONTROLLER_HAS_STALL_DETECTION
+	stepper_motor_controller_get_stall_status_t stepper_motor_get_stall_status;
+
+	stepper_motor_controller_enable_stall_detection_t
+		stepper_motor_controller_enable_stall_detection;
+	stall_detection_motor_controller_set_stall_detection_threshold_t
+		stepper_motor_controller_set_stall_detection_threshold;
+#endif /* CONFIG_STEPPER_MOTOR_CONTROLLER_HAS_STALL_DETECTION */
+#ifdef CONFIG_STEPPER_MOTOR_CONTROLLER_HAS_ABSOLUTE_POSITIONING
+	stepper_motor_controller_set_reference_t stepper_motor_controller_set_reference;
+	stepper_motor_controller_get_position_t stepper_motor_controller_get_position;
+	stepper_motor_controller_set_position_t stepper_motor_controller_set_position;
+#endif /* CONFIG_STEPPER_MOTOR_CONTROLLER_HAS_ABSOLUTE_POSITIONING */
+};
+
+__syscall int32_t stepper_motor_controller_enable(const struct device *dev, bool enable);
+
+static inline int32_t z_impl_stepper_motor_controller_enable(const struct device *dev, bool enable)
+{
+	const struct stepper_motor_controller_api *api =
+		(const struct stepper_motor_controller_api *)dev->api;
+	return api->stepper_motor_controller_enable(dev, enable);
+}
+
+__syscall int32_t stepper_motor_controller_set_steps_to_move(const struct device *dev,
+							     int32_t micro_steps);
+
+static inline int32_t z_impl_stepper_motor_controller_set_steps_to_move(const struct device *dev,
+									int32_t micro_steps)
+{
+	const struct stepper_motor_controller_api *api =
+		(const struct stepper_motor_controller_api *)dev->api;
+	return api->stepper_motor_controller_set_steps_to_move(dev, micro_steps);
+}
+
+__syscall int32_t stepper_motor_controller_set_velocity(const struct device *dev,
+							uint32_t micro_steps_per_sec);
+
+static inline int32_t z_impl_stepper_motor_controller_set_velocity(const struct device *dev,
+								   uint32_t micro_steps_per_sec)
+{
+	const struct stepper_motor_controller_api *api =
+		(const struct stepper_motor_controller_api *)dev->api;
+	return api->stepper_motor_controller_set_velocity(dev, micro_steps_per_sec);
+}
+
+__syscall int32_t stepper_motor_controller_set_micro_step_res(
+	const struct device *dev, enum micro_step_resolution resolution);
+
+static inline int32_t
+z_impl_stepper_motor_controller_set_micro_step_res(const struct device *dev,
+						   enum micro_step_resolution resolution)
+{
+	const struct stepper_motor_controller_api *api =
+		(const struct stepper_motor_controller_api *)dev->api;
+	return api->stepper_motor_controller_set_micro_step_res(dev, resolution);
+}
+
+__syscall int32_t stepper_motor_controller_enable_velocity_mode(const struct device *dev,
+								enum direction dir);
+
+static inline int32_t z_impl_stepper_motor_controller_enable_velocity_mode(const struct device *dev,
```

Comment:
I would not seperate enabling velocity mode and setting the speed into two APIs, simply set the constant velocity to 0 to disable it, and set it to a positive or negative speed to enable it implicitly

Reply:
Hi @bjarki-trackunit I have incorporated your suggesion in the latest update